### ISSUE TITLE
bug(beta): breaking change in variable assignment using environment variables

### ISSUE THREAD

**Issue by nqb** [ovh/venom#349]
On 1.0.0-beta.2, if you assign value to Venom variables using environment variables with following syntax:

```yaml
timezone: "{{.TIMEZONE}}"
```

and use `TIMEZONE=value venom run testsuite`, it doesn't work anymore.

### Steps to reproduce

1. Create a testsuite `env_vars.yml`:

```yml
name: Environment variables
version: 2
vars:
  timezone: "{{.TIMEZONE}}"
testcases:
- name: echo_timezone
  steps:
  - type: exec
    script: "echo {{.timezone}}"
    info: "{{.timezone}}"
```
2. Run `TIMEZONE=UTC venom run env_vars.yml`

=> `Missing variables [TIMEZONE]`

=> `VENOM_VAR_TIMEZONE=UTC venom run env_vars.yml` works but you need to rename all your environment variables.

### Expected results
Keep backward compatibility with previous Venom versions.

**Comment by yesnault** [comment]
It works as expected, cf. https://github.com/ovh/venom#environment-variables

you needs to prefix environment variable with `VENOM_VAR_`.
btw, if you really want to use environment variable in your steps, it's always possible to use them with `script: "echo $TIMEZONE`

**Comment by yesnault** [comment]
about the breaking change: I recheck and add it to the changelog

**Comment by nqb** [comment]
Hello @yesnault,

At the moment, my use case is really what I described in #350: I defined several Venom variables in a global `vars.yml` that used values from environment. Then I use my Venom variables everywhere in several testsuites.

It allows me to update only my variables in `vars.yml ` in place of changing environment variables in every test suite.

**Important**: I just realized that it was a behavior which doesn't require to pass `--env` as Venom argument (as described in [README.md](https://github.com/ovh/venom/blame/v0.28.0/README.md#L22)). It means that following code work on v0.28.0:

File `env_vars_0_28.yml`:
```yml
name: Environment variables
version: 2
testcases:
- name: echo_timezone
  steps:
  - type: exec
    script: "echo {{.TIMEZONE}} > /tmp/timezone.log"
```

`TIMEZONE=UTC venom run env_vars_0_28.yml`

=> `/tmp/timezone.log` contains `UTC` even if we don't pass `--env` as a Venom argument

To summarize: I rely on an implicit behavior (or bug?) of Venom which make possible to use any environment variable without passing `--env` to Venom. So it's normal there is a breaking change.

**Comment by yesnault** [comment]
so, I confirm, it's a breaking change and we don't want anymore to let venom use an environment variable not expected. You have to define somewhere all variables (with format `{{...}}` used), or use directly `echo $TIMEZONE`

**Comment by nqb** [comment]
Thanks for clarification. I will now redefine environment variable names to `VENOM_VAR_MYVAR`

Could you add a note in Changelog of v1.0.0 related to this change ?

**Comment by yesnault** [comment]
> Thanks for clarification. I will now redefine environment variable names to `VENOM_VAR_MYVAR`
> 
> Could you add a note in Changelog of v1.0.0 related to this change ?

Changelog updated
